### DISCUSSION TITLE
MIDI2LR doesn't open anymore

### DISCUSSION THREAD

**Discussion by geld0r** [rsjaffe/MIDI2LR#591] [Problems]
I've been using Midi2Lr with a Behringer X-Touch Mini successfully for a while. Since some time (not sure exactly when it started), it's no longer working. Midi2Lr doesn't start anymore. Neither automatically with Lightroom, nor when explicitly reloading it from Lightroom's plugin menu.

I've tried both reinstalling version 5.2.0.0 and downgrading to 5.1.0.0  without success. The installer complains about not being able to connect to the relevant ports, but nothing is running there and I think the warning was shown with previous updates as well.

I couldn't find any logfile or anything else that indicates what the problem actually is. Any ideas?

**Comment by rsjaffe** [comment]
If you get that installation message and are not simultaneously running MIDI2LR during installation, you have a firewall problem. Check your firewall. MIDI2LR uses ports 58763 and 58764 on localhost (127.0.0.1) to communicate between the application and the plugin.

The plugin leaves a brief log file that's useful when the app doesn't start. The file is _MIDI2LRinfo.txt_ and is in %AppData%\MIDI2LR (for Windows users) or ~/Library/Logs/MIDI2LR (MacOS users).

**Comment by geld0r** [comment]
I've added a rule for the mentioned ports (TCP). I didn't make changes to the firewall/network setup recently though, so not sure if that is the cause.
The config:
![grafik](https://user-images.githubusercontent.com/10269508/146652920-ed0a3478-c263-4a82-9200-66180d3fd062.png)

The logfile reads:

> 
> Log started: 18 Dec 2021 8:00:01pm
> 
> 2021-12-18T20:00:01.886+01:00:C:\Users\rsjaf\source\repos\MIDI2LR\src\application\CommandSet.cpp:G MenuTrans.xml archive loaded from C:\Users\Martin\AppData\Roaming\MIDI2LR\MenuTrans.xml.
> 2021-12-18T20:00:01.894+01:00:C:\Users\rsjaf\source\repos\MIDI2LR\src\application\Main.cpp:ƃ Exception Could not detect cereal root node - likely due to empty or invalid input
> 2021-12-18T20:00:01.894+01:00:C:\Users\rsjaf\source\repos\MIDI2LR\src\application\Main.cpp:Ü Exception Could not detect cereal root node - likely due to empty or invalid input
> 2021-12-18T20:00:01.894+01:00:C:\Users\rsjaf\source\repos\MIDI2LR\src\application\Main.cpp: Terminate called, exception Could not detect cereal root node - likely due to empty or invalid input.

**Reply by rsjaffe** [reply]
Given those errors with cereal, please try the following:
1. While running Lightroom, Select the menu item File -> Plugin Extras -> MIDI2LR -> Delete settings
2. Restart Lightroom
If that doesn't fix it, try the following:
1. Delete the directory %AppData%\MIDI2LR and delete the MIDI2LR.lrplugin directory.
2. Reinstall MIDI2LR.

**Reply by geld0r** [reply]
Step 1 & 2 weren't enough, but deleting the folders and doing a complete reinstall worked.

Maybe it would help for self-diagnosis if the Midi2LR logfile would show up in the plugin extras menu? There is a button to save diagnostic data to file, but it's greyed out for Midi2Lr

Anyways, the cause and resolution are now documented here for others to find. Thank you!
